eng/pipelines/update-azure-linux-pipeline.yml

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# For info about runtime parameters, see https://github.com/microsoft/go-infra/blob/main/docs/pipeline-yml-style.md#runtime-parameters
+parameters:
+  # Allow disabling specific parts of the internal build release. These can be used to re-run
+  # specific parts of a release if necessary, e.g. if a step silently failed.
+  - name: runCreatePullRequest
+    displayName: Create Pull Request on Azure Linux repo
+    type: boolean
+    default: true
+
+  # This parameter intentionally has no default: see release-go-pipeline.yml
+  - name: goReleaseConfigVariableGroup
+    displayName: '[Use "go-release-config" for a real release] Variable group that specifies release target locations and auth.'
+    type: string
+
+  - name: pipelineBuildID
+    displayName: 'The ID of the microsoft-go build pipeline that triggered this pipeline.'
+    type: string
+
+trigger: none
+pr: none
+
+variables:
+  - group: Microsoft-GoLang-bot
+  # Import config group. This may direct the build to use secrets from the other groups.
+  - group: ${{ parameters.goReleaseConfigVariableGroup }}
+resources:
+  repositories:
+    - repository: 1ESPipelineTemplates
+      type: git
+      name: 1ESPipelineTemplates/1ESPipelineTemplates
+      ref: refs/tags/release
+
+extends:
+  template: v1/1ES.Official.PipelineTemplate.yml@1ESPipelineTemplates
+  parameters:
+    sdl:
+      sourceAnalysisPool:
+        name: NetCore1ESPool-Svc-Internal
+        image: 1es-ubuntu-2004
+        os: linux
+      suppression:
+        suppressionFile: $(Build.SourcesDirectory)/.config/guardian/.gdnsuppress
+      tsa:
+        enabled: true
+        configFile: $(Build.SourcesDirectory)/.config/tsa/tsaoptions.json
+
+    stages:
+      - stage: Release
+        jobs:
+          - template: /eng/pipelines/jobs/releasego.yml@self
+            parameters:
+              steps:
+                # Download build asset JSON from completed internal build to update Azure Linux specfiles with. 
+                - task: DownloadPipelineArtifact@2 
+                  displayName: Download Build Asset JSON 
+                  inputs: 
+                    source: specific 
+                    artifact: BuildAssets 
+                    project: internal 
+                    pipeline: $(GoPipelineID) 
+                    runVersion: specific 
+                    runId: ${{ parameters.pipelineBuildID }}
+                    path: $(assetsDir)
+
+                - ${{ if eq(parameters.runCreatePullRequest, true) }}:
+                  - script: |
+                      releasego update-azure-linux \
+                        -pat '$(GitHubPAT)'
+                        -build-asset-json '$(buildAssetJsonFile)'
+                    displayName: Create Pull Request to Update Azure Linux Specfiles
```

Comment:
In the example pipeline run https://dev.azure.com/dnceng/internal/_build/results?buildId=2525925&view=results, this step doesn't show up--has it been run?

I've cloned go-release-config-dagood to https://dev.azure.com/dnceng/internal/_library?itemType=VariableGroups&view=VariableGroupView&variableGroupId=198&path=go-release-config-mertakman for you to use for testing, if that helps. `go-release-config-mertakman` lets you set your own GitHubPAT as a secret. I would recommend making a bot GitHub account to make a simple access boundary--then, at worst, any problem is just a nuisance. Once you've saved that, passing `go-release-config-mertakman` to `goReleaseConfigVariableGroup` lets you do more in depth test runs with a custom GitHubPAT.

Adding runtime parameters is also fine for testing. (E.g. for target repo organization/name.) I don't mind leaving them in the final yml to make future testing easier too. Parameters are simpler than variables (and variable groups), so I'd recommend parameters when it's reasonable. It's not possible to make a "secret" parameter, so for GitHubPAT the variable group is necessary.

Reply:
The reason the step in question didn't run is that the parameter Create Pull Request on Azure Linux repo was not set to true for this pipeline run. By default, this parameter is off, and hence the associated step was skipped during execution.

Reply:
added.